### PR TITLE
refactor to use scattered reads and reusable buffers

### PR DESCRIPTION
**Pull request by kortemik**
refactor to use scattered reads and reusable buffers

### CODE REVIEW

File: src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java
Comment (eemhu) [inline src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java:1]: empty on purpose?

File: src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java
Comment (eemhu) [inline src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java:1]: this also empty on purpose?

File: src/main/java/com/teragrep/rlp_03/FrameProcessor.java
Comment (eemhu) [inline src/main/java/com/teragrep/rlp_03/FrameProcessor.java:1]: -or ending objects clash with object thinking principles, but perhaps that is for another time?

**Review by eemhu — APPROVED** [review]
Seems fairly good to me; made some comments though.

File: src/main/java/com/teragrep/rlp_03/FrameProcessor.java
Comment (kortemik) [reply src/main/java/com/teragrep/rlp_03/FrameProcessor.java:1]: for another time, it's existing thing

File: src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java
Comment (kortemik) [reply src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java:1]: need to create this, forgot to, thanks!

File: src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java
Comment (kortemik) [reply src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java:1]: need to create this too, forgot to, thanks!

File: src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java
Comment (kortemik) [reply src/test/java/com/teragrep/rlp_03/context/frame/access/AccessTest.java:1]: fixed in e954031d50ea34ad430e534fea6d1e85b65a3f16

File: src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java
Comment (kortemik) [reply src/test/java/com/teragrep/rlp_03/context/frame/fragment/FragmentTest.java:1]: implemented via dbee974d0307426e29ee47f2d093ba4d7b41d4f1

**Comment by eemhu** [comment]
added tests lgtm

**Pull request merged**